js/ui/viewSelector.js

```diff
@@ -132,6 +139,16 @@ const ViewsDisplayLayout = new Lang.Class({
         let heightAboveGrid = this._calcAllViewPlacement(allViewHeight, entryHeight, availHeight);
         this._heightAboveEntry = this._centeredHeightAbove(entryHeight, heightAboveGrid);
 
+        let gcHeight = this._gc.get_preferred_height(availWidth)[1];
+
+        let gcBox = box.copy();
+        let x1 = (availWidth - this._gc.get_width()) * 0.5;
+        gcBox.y1 = this._heightAboveGC;
+        gcBox.y2 = gcBox.y1 + gcHeight;
+        gcBox.x1 = x1;
+        gcBox.x2 = x1 + this._gc.get_width();
+        this._gc.allocate(gcBox, flags);
```

Comment:
While this is consistent with the existing code, my preference here would be to create a subclass of `St.Button` for the discovery feed handle and encapsulate as much of its code as possible inside of instance methods. We would only need to pass `availWidth` and `this._heightAboveGC` to it.